applets/notification_area/na-grid.c

```diff
@@ -128,52 +236,32 @@ item_removed_cb (NaHost *host,
 
   gtk_container_remove (GTK_CONTAINER (self), GTK_WIDGET (item));
   self->items = g_slist_remove (self->items, item);
+  refresh_grid (self);
 }
 
 static void
-update_size_and_orientation (NaGrid         *self,
-                             GtkOrientation  orientation)
+na_grid_size_allocate (GtkWidget     *grid,
+                       GtkAllocation *allocation,
+                       gpointer       data)
 {
-  /* FIXME: do we really need that?  comes from NaTray */
-  /* FIXME: if we do, do that in overridden preferred size handlers */
-
-  /* note, you want this larger if the frame has non-NONE relief by default. */
-  switch (orientation)
-    {
-    case GTK_ORIENTATION_VERTICAL:
-      /* Give grid a min size so the frame doesn't look dumb */
-      gtk_widget_set_size_request (GTK_WIDGET (self), MIN_GRID_SIZE, -1);
-      break;
-    case GTK_ORIENTATION_HORIZONTAL:
-      gtk_widget_set_size_request (GTK_WIDGET (self), -1, MIN_GRID_SIZE);
-      break;
-    }
-}
-
-static void
-orientation_notify (GObject    *object,
-                    GParamSpec *pspec,
-                    gpointer    data)
-{
-  update_size_and_orientation (NA_GRID (object),
-                               gtk_orientable_get_orientation (GTK_ORIENTABLE (object)));
+  refresh_grid (NA_GRID (grid));
 }
 
 static void
 na_grid_init (NaGrid *self)
 {
-  GtkOrientation orientation;
-
   self->icon_padding = 0;
   self->icon_size = 0;
 
   self->hosts = NULL;
   self->items = NULL;
+  self->grid_props = g_new0 (GridProps, 1);
+  self->min_icon_size = MIN_ICON_SIZE_DEFAULT;
+  
+  gtk_grid_set_row_homogeneous (GTK_GRID (self), TRUE);
+  gtk_grid_set_column_homogeneous (GTK_GRID (self), TRUE);
 
-  orientation = gtk_orientable_get_orientation (GTK_ORIENTABLE (self));
-  update_size_and_orientation (self, orientation);
-
-  g_signal_connect (self, "notify::orientation", G_CALLBACK (orientation_notify), NULL);
+  g_signal_connect (self, "size-allocate", G_CALLBACK (na_grid_size_allocate), NULL);
```

Comment:
I would prefer if this was `widget_class->size_allocate = ...;`, but that should not change functionality - so whatever MATE developers (I am not) prefer.

Reply:
Did you chain-up? `GTK_WIDGET_CLASS (na_grid_parent_class)->size_allocate (grid, allocation);`?

Reply:
No, now I know what's the difference between the g_signal_connect and setting it on the class. Thanks